eng/emitter-package.json

```diff
@@ -0,0 +1,16 @@
+{
+  "main": "dist/src/index.js",
+  "dependencies": {
+    "@azure-tools/cadl-python": "0.4.25",
+    "@autorest/python": "6.4.0",
```

Comment:
@msyyc can we change this to just have the cadl-python dependency? Like .net has? https://github.com/Azure/azure-sdk-for-net/blob/main/eng/emitter-package.json

Reply:
@iscai-msft  @azure-tools/cadl-python depends on latest autorest.python by default. So if autorest.python has a update with breaking, it may break @azure-tools/cadl-python.

Reply:
we should catch that in our tests then because we always test our cadl-python on the latest autorest.python. If it breaks, we can add tests to ensure this doesn't happen. 

It's also not good politically-speaking exposing all of these dependencies, especially the one on autorest because we don't want to show that our cadl emitter is depending on our autorest emitter, it's just an implementation detail for now.

Reply:
The first target of pipeline is stale. Typespec/emitters are not stable enough and that's why we pin almost all the dependencies.

Reply:
I agree with @iscai-msft that we should try to reduce this set and if there are specific dependencies needed they should be added to the python emitter package.